### ISSUE TITLE
@gridsome/source-contentful: Optional asset fields don't appear in the schema unless they all are not empty

### ISSUE THREAD

**Issue by weaversam8** [gridsome/gridsome#864]
## Description

If I create a Content Type that has an asset field that's optional, it won't appear in the GraphQL schema unless it is given a value on every Content entry.

### Steps to reproduce

1. Create a new Gridsome project and a Contentful project to test with, and configure the @gridsome/source-contentful plugin.
2. Add a new type to the Contentful project that has an asset field that isn't required.
3. Add a Content entry in the Contentful project with an asset in the asset field.
3. Start the Gridsome project and open the GraphQL explorer. Open the "schema" tab.
4. You'll notice that your asset field is listed on your Contentful(TypeName) schema.
5. Add another Content entry with no asset in the asset field.
6. Restart the Gridsome project.

### Expected result

The asset field should still be in the schema for the ContentfulXXX type.

### Actual result

The asset field is no longer in the schema.

### Environment

```
  System:
    OS: Windows 10
    CPU: (4) x64 Intel(R) Core(TM) i7-6600U CPU @ 2.60GHz
  Binaries:
    Node: 10.15.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.12.3 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.4.1 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: 44.17763.831.0
  npmPackages:
    gridsome: ^0.7.0 => 0.7.8
```

I'm using version `^0.5.1 => 0.5.1` of @gridsome/source-contentful.

**Comment by weaversam8** [comment]
So, after some further investigation, this is what's happening:

Contentful's API only returns the fields that are set when querying an entry. This means that in the above example, you might query the API and get two responses like this:
```js
// For the first entry, which has the asset field set
{
   field1: "value1",
   field2: "value2",
   assetField: { ... }
}
// But for a second entry, which doesn't have a value set
{
   field1: "value3",
   field2: "value4"
}
```

This is problematic. If Contentful returned something like `assetField: null` in the response, we wouldn't have this problem.

Currently, [we're adding entries to the store based on the fields we receive from Contentful](https://github.com/gridsome/gridsome/blob/55695c27f79d5cdcf6974f5d1e17b2202a1c5f11/packages/source-contentful/index.js#L87). This means that if we're missing a field, it just doesn't go into the store. Somewhere in the process, this breaks our GraphQL schema and the field is omitted entirely. If we iterate based on the fields we know _should_ exist (by referencing the schema we already have) then we'll catch the missing fields and give them an undefined value properly.

I've fixed this locally and am working on a PR right now.

**Comment by weaversam8** [comment]
PR is now passing CI and waiting for merge. Not sure what I need to do w/r/t a version bump if this would be a new version of [@gridsome/source-contentful](https://www.npmjs.com/package/@gridsome/source-contentful)... cc @hjvedvik.

**Comment by weaversam8** [comment]
As a temporary workaround until #866 is merged, you can add the following line to your `package.json` to use the patched version of the package:

```js
"@gridsome/source-contentful": "github:weaversam8/gridsome-source-contentful#issue-864-fix"
```

**Comment by weaversam8** [comment]
@hjvedvik when can we expect an patched version of `@gridsome/source-contentful` to be published to NPM with this fix?

**Comment by weaversam8** [comment]
Update, the patched version (0.5.2) of [`@gridsome/source-contentful`](https://www.npmjs.com/package/@gridsome/source-contentful) was published yesterday in [`55dc559`](https://github.com/gridsome/gridsome/commit/55dc559e9ec7072042cb512b5d5defb7ef7360cb). Thanks Hans!
